tests/e2e/testcases/node/monitoring/suite.go

```diff
@@ -0,0 +1,208 @@
+		gomega.Expect(clustersConfig.Clusters[constants.E2EClusterName].Network.Kind.String()).To(gomega.Equal(networkCapitalized))
+		gomega.Expect(clustersConfig.Clusters[constants.E2EClusterName].Nodes).To(gomega.HaveLen(numNodes))
+		monitoringHostID = clustersConfig.Clusters[constants.E2EClusterName].MonitoringInstance
+		createdHostsFormatted = append(createdHostsFormatted, clustersConfig.Clusters[constants.E2EClusterName].Nodes...)
+	})
+	ginkgo.It("checks prometheus config in monitoring host", func() {
+		usr, err := user.Current()
+		gomega.Expect(err).Should(gomega.BeNil())
+		homeDir := usr.HomeDir
+		monitoringHost, err := ansible.GetInventoryFromAnsibleInventoryFile(filepath.Join(homeDir, constants.BaseDirName, relativePath, constants.AnsibleInventoryDir, "e2e", "monitoring"))
+		gomega.Expect(err).Should(gomega.BeNil())
+		err = commands.DownloadPrometheusConfig(monitoringHost[0], filepath.Join(homeDir, constants.BaseDirName, relativePath, monitoringHostID))
+		gomega.Expect(err).Should(gomega.BeNil())
+		createdDockerHosts, err := ansible.GetInventoryFromAnsibleInventoryFile(filepath.Join(homeDir, constants.BaseDirName, relativePath, constants.AnsibleInventoryDir, "e2e"))
+		gomega.Expect(err).Should(gomega.BeNil())
+		createdHosts = createdDockerHosts
+		hostavalancheGoPorts := []string{}
+		hostMachinePorts := []string{}
+		for _, host := range createdHosts {
+			hostavalancheGoPorts = append(hostavalancheGoPorts, fmt.Sprintf("%s:9650", host.IP))
+			hostMachinePorts = append(hostMachinePorts, fmt.Sprintf("%s:9100", host.IP))
+		}
+		type StaticConfig struct {
+			Targets []string `yaml:"targets"`
+		}
+		type ScrapeConfig struct {
+			JobName       string         `yaml:"job_name"`
+			StaticConfigs []StaticConfig `yaml:"static_configs"`
+		}
+		type PrometheusConfig struct {
+			ScrapeConfigs []ScrapeConfig `yaml:"scrape_configs"`
+		}
+		data, err := os.ReadFile(filepath.Join(homeDir, constants.BaseDirName, relativePath, monitoringHostID, constants.NodePrometheusConfigFileName))
+		gomega.Expect(err).Should(gomega.BeNil())
+		var prometheusConfig PrometheusConfig
+		err = yaml.Unmarshal(data, &prometheusConfig)
+		gomega.Expect(err).Should(gomega.BeNil())
+		scrapeConfig := prometheusConfig.ScrapeConfigs
+		avalancheGoJob := "avalanchego"
+		avalancheGoMachineJob := "avalanchego-machine"
+		for _, newConfig := range scrapeConfig {
+			if newConfig.JobName == avalancheGoJob || newConfig.JobName == avalancheGoMachineJob {
+				targets := newConfig.StaticConfigs
+				dockerTarget := targets[0]
+				gomega.Expect(len(dockerTarget.Targets)).To(gomega.Equal(numNodes))
+				if newConfig.JobName == avalancheGoJob {
+					for _, host := range hostavalancheGoPorts {
+						gomega.Expect(slices.Contains(dockerTarget.Targets, host)).To(gomega.Equal(true))
+					}
+				} else {
+					for _, host := range hostMachinePorts {
+						gomega.Expect(slices.Contains(dockerTarget.Targets, host)).To(gomega.Equal(true))
+					}
+				}
+			}
+		}
+	})
```

Comment:
you need to check that all monitored hosts have prometheus agents installed and ideally running by accessing their exposed metrics